test/integration/sapi-nv-policy-locality.int.c

```diff
@@ -0,0 +1,311 @@
+#include "util/aux_util.h"
+#define LOGMODULE test
+#include "util/log.h"
+
+#define NV_INDEX 0x01800003
+#define NV_SIZE 96
+
+#define INIT_SIMPLE_TPM2B_SIZE(type) (type).size = sizeof(type) - 2;
+#define TPM2B_SIZE_MAX(type) (sizeof (type) - 2)
+
+/*
+ * This test creates an NV index governed by a policy and then performs
+ * several operations on the NV region to exercise this policy. The NV
+ * region created is modeled after the TXT AUX region as defined by the
+ * Intel TXT software developers guide:
+ * https://www.intel.com/content/dam/www/public/us/en/documents/guides/intel-txt-software-development-guide.pdf
+ * Read is controlled by authValue and is unrestricted since authValue is
+ * set to emptyBuffer.
+ * Write is controlled by policy that allows writes from locality 3 and 4.
+ */
+/*
+ * This function creates a policy session asserting that the locality is
+ * either 3 or 4. If this policy is used when executing a command and the
+ * policy is not satisfied (locality is not 3 or 4) then the command will
+ * fail.
+ */
+static TSS2_RC
+create_policy_session (TSS2_SYS_CONTEXT *sys_ctx,
+                       TPMI_SH_AUTH_SESSION *handle)
+{
+    TPMA_LOCALITY locality = TPMA_LOCALITY_TPM2_LOC_THREE |
+        TPMA_LOCALITY_TPM2_LOC_FOUR;
+    TPM2B_NONCE nonce = { .size = GetDigestSize (TPM2_ALG_SHA1), };
+    TPM2B_NONCE nonce_tpm = { .size = TPM2B_SIZE_MAX (nonce_tpm), };
+    TSS2_RC rc;
+    TPM2B_ENCRYPTED_SECRET salt = { 0, };
+    TPMT_SYM_DEF symmetric = { .algorithm = TPM2_ALG_NULL, };
+
+    rc = Tss2_Sys_StartAuthSession (sys_ctx,
+                                    TPM2_RH_NULL,
+                                    TPM2_RH_NULL,
+                                    0,
+                                    &nonce,
+                                    &salt,
+                                    TPM2_SE_POLICY,
+                                    &symmetric,
+                                    TPM2_ALG_SHA1,
+                                    handle,
+                                    &nonce_tpm,
+                                    0);
+    return_if_error (rc, "Tss2_Sys_StartAuthSession");
+
+    rc = Tss2_Sys_PolicyLocality (sys_ctx,
+                                  *handle,
+                                  0,
+                                  locality,
+                                  0);
+    return_if_error (rc, "Tss2_Sys_PolicyLocality");
+    return rc;
+}
+/*
+ * This function creates the NV region used in this test. The appropriate
+ * attributes are applied using the nvPublic member of the TPM2B_NV_PUBLIC
+ * structure.
+ */
+static TSS2_RC
+setup_nv (TSS2_SYS_CONTEXT *sys_ctx)
+{
+    TSS2_RC rc;
+    TPMI_SH_AUTH_SESSION auth_handle = 0;
+    TPM2B_DIGEST policy_hash = { .size = TPM2B_SIZE_MAX (policy_hash), };
+    TPM2B_AUTH nv_auth = { 0, };
+    TSS2L_SYS_AUTH_RESPONSE auth_rsp = { 0, };
+    TPM2B_NV_PUBLIC public_info = {
+        .size = sizeof (TPMI_RH_NV_INDEX) + sizeof (TPMI_ALG_HASH) +
```

Comment:
.size = 0 for consistency. See #1353

Reply:
Good catch. Forgot tss2-sys does these calcs for us now. This code is ancient & predates that change.